e2etest/pkg/status/l2status.go

```diff
@@ -5,36 +5,40 @@ package status
 import (
 	"context"
 	"fmt"
+	"go.universe.tf/e2etest/pkg/metallb"
 
 	"go.universe.tf/metallb/api/v1beta1"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetL2Status(cs client.Client, svc *v1.Service, nodeName string) (*v1beta1.ServiceL2Status, error) {
-	s := &v1beta1.ServiceL2Status{}
-	if err := cs.Get(context.TODO(), types.NamespacedName{
-		Namespace: svc.Namespace,
-		Name:      fmt.Sprintf("%s-%s", svc.Name, nodeName),
-	}, s); err != nil {
+const (
+	LabelServiceName      = "metallb.io/service-name"
+	LabelServiceNamespace = "metallb.io/service-namespace"
+)
+
+func L2ForService(cs client.Client, svc *v1.Service) (*v1beta1.ServiceL2Status, error) {
+	statuses, err := L2sForService(cs, svc)
+	if err != nil {
 		return nil, err
 	}
-	return s, nil
+	if len(statuses) != 1 {
+		return nil, fmt.Errorf("expected 1 L2 status, got %d", len(statuses))
+	}
+	return &(statuses[0]), nil
 }
 
-func GetSvcPossibleL2Status(cs client.Client, svc *v1.Service, nodes *v1.NodeList) ([]*v1beta1.ServiceL2Status, error) {
-	var l2Statuses []*v1beta1.ServiceL2Status
-	for _, node := range nodes.Items {
-		s, err := GetL2Status(cs, svc, node.Name)
-		if err != nil && errors.IsNotFound(err) {
-			continue
-		}
-		if err != nil {
-			return nil, err
-		}
-		l2Statuses = append(l2Statuses, s)
+func L2sForService(cs client.Client, svc *v1.Service) ([]v1beta1.ServiceL2Status, error) {
```

Comment:
nit: this is a bit confusing, we can make this private or move the logic to the `L2ForService`, and create another function (e.g `L2StatusExistsFor`) for the other place you determine if a status exists or not.

Reply:
We can just use L2ForService as in every place we use L2sForServices we error if we get more than one entry

Reply:
There is one single case we need to validate there's 0 status obj and we can't ensure this from the error returned by L2ForService. So I had to expose L2sForService.